### DISCUSSION TITLE
`sf org create user --definition-file` could be more complete, to allow assigning permission set licenses to the created user

### DISCUSSION THREAD

**Discussion by RupertBarrow** [forcedotcom/cli#2889] [Ideas]
I'm creating an API-only integration test user to run automated integration tests in a scratch org in my CI pipeline.

My user needs the following features : 
- profile = "Minimum Access - API Only Integrations"
- permission set = "my minimum permissions"
- and I discovered it also needed : permission set license = "Salesforce API Integration"

Please let me set permission set licenses in the `--definition-file` file, just like profile and permission sets.
This would avoid me having to write the ugly 
```
username=$(sf org create user --target-org org --definition-file config/user.json --json | jq -r '.result.fields.username')
sf org assign permsetlicense --name "Salesforce API Integration" --on-behalf-of ${username} --target-org org
```

**Comment by mshanemc** [comment]
not an argument against the feature request, but FYI you *can* use aliases for users, like

```
sf org create user --set-alias otherUser ...
sf org assign permsetlicense --on-behalf-of otherUser ...
```

**Reply by RupertBarrow** [reply]
> not an argument against the feature request, but FYI you _can_ use aliases for users, like
> 
> ```
> sf org create user --set-alias otherUser ...
> sf org assign permsetlicense --on-behalf-of otherUser ...
> ```

Right.
But I can't use that alias to `sf org login jwt --username otherUser ...`
I still need to grab the real username to pass to `sf org login jwt --username`

**Comment by RupertBarrow** [comment]
I've just discovered that the case I was demonstrating above actually doesn't event work : to assign "my minimum permissions" permset to the user, the API Only user must *already* have the "Salesforce API Integration" permset license assigned.

So, from a feature request, this becomes a double-whammy bug report : 
1) I cannot assign a permset with --definition-file to an API Only user because the permset license needs to be assigned first
2) When I do so, the CLI returns neither an error nor a warning; the user is created but without it's permset
